src/libraries/System.Private.CoreLib/src/System/IO/Enumeration/FileSystemEntry.Windows.cs

```diff
@@ -75,5 +75,18 @@ public FileSystemInfo ToFileSystemInfo()
         /// <returns>A string representing the full path.</returns>
         public string ToFullPath() =>
             Path.Join(Directory, FileName);
+
+        private static string Join(
+            ReadOnlySpan<char> originalRootDirectory,
+            ReadOnlySpan<char> relativePath,
+            ReadOnlySpan<char> fileName)
+        {
+            if (originalRootDirectory.Length == 2 && originalRootDirectory[1] == Path.VolumeSeparatorChar)
```

Comment:
Is it possible for a Windows drive to have more than one letter?

Reply:
When A-Z alphabets are all assigned, user either needs to switch to UNC paths via GUIDs `\\?\Volume{GUID}` to access the next device, or mount it as directory (via Disk management). That device would not show up in explorer by default.